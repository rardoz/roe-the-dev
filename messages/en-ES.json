{
  "Home": {
    "title": "Roe the Dev. Hola!"
  }
}
