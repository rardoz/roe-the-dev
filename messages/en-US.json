{
  "Home": {
    "title": "Roe the Dev"
  },
  "Nav": {
    "portfolio": "Porfolio",
    "blog": "Blog",
    "experiments": "Experiments",
    "menuIconAltText": "Click for additional menu links",
    "logoAltText": "Roe the Dev logo",
    "about": "About Roe"
  },
  "About": {
    "title": "About Roe",
    "subtitle": "Tech Lead, Software Engineer, Mentor And Artist",
    "description1": "I am a versatile digital nomad and innovative creative technologist assuming the role of a tech lead while embracing a nomadic lifestyle, dedicated to bringing creative ideas to fruition.",
    "description2": "I pursued my education at Black Hawk College and The Illinois Institute of Art in Chicago, achieving an associate's degree in visual communications. With a professional tenure of over a decade, I have accumulated substantial experience in software development.",
    "cta": "Learn more"
  },
  "Portfolio": {
    "title": "Portfolio",
    "cta": "See my work",
    "footerMessage": "Background AI generated with"
  },
  "Blog": {
    "title": "From my desk",
    "cta": "Read blogs",
    "footerMessage": "Background AI generated with"
  }
}
