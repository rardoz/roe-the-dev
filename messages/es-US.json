{
  "Home": {
    "title": "Roe the Dev. Hola!"
  },
  "Nav": {
    "portfolio": "Cartera",
    "blog": "Blog",
    "experiments": "Experimentación",
    "menuIconAltText": "Haga clic para ver más enlaces",
    "logoAltText": "Logotipo de Roe the Dev",
    "about": "Sobre Roe"
  },
  "About": {
    "title": "Sobre Roe",
    "subtitle": "Líder Tecnológico, Ingeniero de Software, Mentor y Artista",
    "description1": "Soy un nómada digital versátil e innovador tecnólogo creativo, asumiendo el rol de líder tecnológico mientras abrazo un estilo de vida nómada, dedicado a llevar a cabo ideas creativas.",
    "description2": "Cursé mis estudios en Black Hawk College y en The Illinois Institute of Art en Chicago, obteniendo un título de asociado en comunicación visual. Con una experiencia profesional de más de una década, he acumulado una considerable experiencia en desarrollo de software.",
    "cta": "Aprender más"
  },
  "Portfolio": {
    "title": "Portafolio",
    "cta": "Ver mi trabajo",
    "footerMessage": "Fondo generado por inteligencia artificial con"
  },
  "Blog": {
    "title": "Desde mi escritorio",
    "cta": "Leer blogs"
  }
}
