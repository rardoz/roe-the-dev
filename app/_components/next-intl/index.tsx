import { createLocalizedPathnamesNavigation } from 'next-intl/navigation'

import config from '../../../messages/config'

export const { Link, redirect, usePathname, useRouter, getPathname } =
  createLocalizedPathnamesNavigation({
    locales: config.locales,
    pathnames: config.pathnames,
  })

import { ComponentProps } from 'react'
import type { AppPathnames } from '../../../messages/config'

export function UnstyledLink<Pathname extends AppPathnames>({
  href,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return <Link href={href} {...rest} />
}

export function StyledLink<Pathname extends AppPathnames>({
  className,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return (
    <UnstyledLink
      {...rest}
      className={`transition-colors duration-500 text-pink-500 hover:text-pink-400 ${
        className || ''
      }`}
    />
  )
}
