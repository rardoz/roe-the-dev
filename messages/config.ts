import { Pathnames } from 'next-intl/navigation'

const localePrefix: 'as-needed' | 'always' | 'never' | undefined = 'as-needed'

export const locales = ['en-US', 'es-US']

export type AppPathname = Pathnames<typeof locales>

export const pathnames = {
  '/blog': '/blog',
  '/blog/[slug]': '/blog/[slug]',
  '/portfolio': '/portfolio',
  '/portfolio/[slug]': '/portfolio/[slug]',
  '/experiment': '/experiment',
  '/': { 'en-US': '/', 'es-US': '/' },
} satisfies AppPathname

export type AppPathnames = keyof typeof pathnames

const config = {
  locales,
  defaultLocale: 'en-US',
  localePrefix,
  pathnames,
}

export default config
