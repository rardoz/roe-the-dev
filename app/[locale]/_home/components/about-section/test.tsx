import { render } from '@testing-library/react'
import { NextIntlClientProvider } from 'next-intl'
import messages from '../../../../../messages/en-US.json'
import AboutSection from '.'

describe('About Section', () => {
  it('should render with default locale set', () => {
    expect(
      render(
        <NextIntlClientProvider locale="en-US" messages={messages}>
          <AboutSection />
        </NextIntlClientProvider>,
      ).container,
    ).toMatchSnapshot()
  })
})
