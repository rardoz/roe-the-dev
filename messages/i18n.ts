import { getRequestConfig } from 'next-intl/server'
import { default as config, locales } from './config'

// export default getRequestConfig(async ({ requestLocale }) => ({
//   messages: (await import(`./${requestLocale}.json`)).default,
// }))

export default getRequestConfig(async ({ requestLocale }) => {
  let _locale = await requestLocale
  if (!_locale || locales.includes(_locale as any)) {
    _locale = config.defaultLocale
  }

  return {
    locale: _locale,
    messages: (await import(`./${_locale}.json`)).default,
  }
})
