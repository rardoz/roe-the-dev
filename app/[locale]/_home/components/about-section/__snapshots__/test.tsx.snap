// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Section should render with default locale set 1`] = `
<div>
  <div
    class="w-3/4 lg:w-1/2 mx-auto mt-8 -mb-4"
  >
    <h2
      class="text-center text-4xl font-bold text-purple-700"
    >
      About Roe
    </h2>
    <hr
      class="mt-5 mb-16 border-dotted border-spacing-8 border-8 border-t-0 border-purple-300"
    />
  </div>
  <div
    class="w-3/4 lg:w-1/2"
  >
    <div
      class="flex justify-center items-start flex-wrap sm:flex-nowrap"
    >
      <div
        class="flex-shrink-0 px-4 mb-4"
      >
        <img
          alt="about"
          class="rounded-full border-8 border-white drop-shadow-lg hover:drop-shadow-xl hover:scale-105 transition-all duration-500 ease-in-out"
          data-nimg="1"
          decoding="async"
          height="250"
          loading="lazy"
          src="/_next/image?url=%2Froe-profile-pic.png&w=640&q=75"
          srcset="/_next/image?url=%2Froe-profile-pic.png&w=256&q=75 1x, /_next/image?url=%2Froe-profile-pic.png&w=640&q=75 2x"
          style="color: transparent;"
          width="250"
        />
        <div
          class=" flex lg:hidden justify-center items-center mt-5"
        >
          <a
            class="bg-purple-700 text-white font-bold py-2 px-4 rounded-full hover:bg-pink-600 transition-all duration-500 ease-in-out"
            href="/en-US/about"
          >
            Learn more
          </a>
        </div>
      </div>
      <div
        class="flex-shrink flex-grow px-4 text-center sm:text-left"
      >
        <h3
          class="font-semibold pb-4 text-lg"
        >
          Tech Lead, Software Engineer, Mentor And Artist
        </h3>
        <p>
          I am a versatile digital nomad and innovative creative technologist assuming the role of a tech lead while embracing a nomadic lifestyle, dedicated to bringing creative ideas to fruition.
        </p>
        <p
          class="my-4"
        >
          I pursued my education at Black Hawk College and The Illinois Institute of Art in Chicago, achieving an associate's degree in visual communications. With a professional tenure of over a decade, I have accumulated substantial experience in software development.
        </p>
        <div
          class="hidden lg:block mt-8"
        >
          <a
            class="bg-purple-700 text-white font-bold py-2 px-4 rounded-full hover:bg-pink-600 transition-all duration-500 ease-in-out "
            href="/en-US/about"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
